supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointInformation.java

```diff
@@ -0,0 +1,118 @@
+package de.tum.cit.endpointanalysis;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class EndpointInformation {
+
+    private String requestMapping;
+
+    private String endpoint;
+
+    private String httpMethodAnnotation;
+
+    private String URI;
+
+    private String className;
+
+    private int line;
+
+    private List<String> otherAnnotations;
+
+    public EndpointInformation() {
+
+    }
+
+    public EndpointInformation(String requestMapping, String endpoint, String httpMethodAnnotation, String URI, String className, int line, List<String> otherAnnotations) {
+        this.requestMapping = requestMapping;
+        this.endpoint = endpoint;
+        this.httpMethodAnnotation = httpMethodAnnotation;
+        this.URI = URI;
+        this.className = className;
+        this.line = line;
+        this.otherAnnotations = otherAnnotations;
+    }
```

Comment:
Couldn't this be a record instead of a class? 

The file should be much simpler then and most likely 1/3 of the current size

Reply:
You are right. I always thought records could not have methods, but that seems incorrect. I will adjust it accordingly.